crates/bevy_light/src/lib.rs

```diff
@@ -686,25 +680,21 @@ pub fn check_point_light_mesh_visibility(
                                 if !**view_visibility {
                                     view_visibility.set();
                                 }
+                                **was_visible_now_hidden = false;
                                 spot_visible_entities_local_queue.push(entity);
                             }
                         } else {
                             if !**view_visibility {
                                 view_visibility.set();
                             }
+                            **was_visible_now_hidden = false;
```

Comment:
I really don't like this inherited footgun that caused the regression. You need to update `ViewVisibility` as well as `WasVisibleNowHidden` to make things work correctly.

We are already doing a table scan over `WasVisibleNowHidden`, I think we should consider making `ViewVisibility::set` private, and only have users interact with `WasVisibleNowHidden`. This would ensure users wouldn't be able to mess up change detection, because our internal systems handle this for users based on the state of `WasVisibleNowHidden`.